bouncerscript/src/bouncerscript/constants.py

```diff
@@ -66,84 +66,124 @@
+        r".*\.(?:bz2|xz|dmg|exe|mar|msi|msix))$"
     ),
     "thunderbird": (
         r"^(/thunderbird/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/(?:\:lang|multi)/"
-        r"(?:thunderbird|Thunderbird).*\.(?:bz2|dmg|exe|mar|msi|msix))$"
+        r"(?:thunderbird|Thunderbird).*\.(?:bz2|xz|dmg|exe|mar|msi|msix))$"
     ),
 }
 
 _BOUNCER_PATH_REGEXES_PER_PRODUCT_DEFAULT = {
     "firefox-nightly-latest": (
-        r"^(/firefox/nightly/latest-(mozilla-central|try)-l10n/firefox-\d+\.0a1\.:lang\."
-        r"(?:linux-i686\.tar\.bz2|linux-x86_64\.tar\.bz2|linux-aarch64\.tar\.bz2|mac\.dmg|win32\.installer\.exe|win64\.installer\.exe|win64-aarch64\.installer\.exe))$"
+        r"^(/firefox/nightly/latest-(mozilla-central|try)-l10n/"
+        r"firefox-\d+\.0a1\.:lang\."
+        r"(?:"
+        r"linux-(i686|x86_64|aarch64)\.tar\.(bz2|xz)|"
+        r"mac\.dmg|"
+        r"win(32|64|64-aarch64)\.installer\.exe"
+        r"))$"
     ),
     "firefox-nightly-latest-ssl": (
-        r"^(/firefox/nightly/latest-(mozilla-central|try)/firefox-\d+\.0a1\.en-US\."
-        r"(?:linux-i686\.tar\.bz2|linux-x86_64\.tar\.bz2|linux-aarch64\.tar\.bz2|mac\.dmg|win32\.installer\.exe|win64\.installer\.exe|win64-aarch64\.installer\.exe))$"
+        r"^(/firefox/nightly/latest-(mozilla-central|try)/"
+        r"firefox-\d+\.0a1\.en-US\."
+        r"(?:"
+        r"linux-(i686|x86_64|aarch64)\.tar\.(bz2|xz)|"
+        r"mac\.dmg|"
+        r"win(32|64|64-aarch64)\.installer\.exe"
+        r"))$"
     ),
     "firefox-nightly-latest-l10n": (
-        r"^(/firefox/nightly/latest-(mozilla-central|try)-l10n/firefox-\d+\.0a1\.:lang\."
-        r"(?:linux-i686\.tar\.bz2|linux-x86_64\.tar\.bz2|linux-aarch64\.tar\.bz2|mac\.dmg|win32\.installer\.exe|win64\.installer\.exe|win64-aarch64\.installer\.exe))$"
+        r"^(/firefox/nightly/latest-(mozilla-central|try)-l10n/"
+        r"firefox-\d+\.0a1\.:lang\."
+        r"(?:"
+        r"linux-(i686|x86_64|aarch64)\.tar\.(bz2|xz)|"
+        r"mac\.dmg|"
+        r"win(32|64|64-aarch64)\.installer\.exe"
+        r"))$"
     ),
     "firefox-nightly-latest-l10n-ssl": (
-        r"^(/firefox/nightly/latest-(mozilla-central|try)-l10n/firefox-\d+\.0a1\.:lang\."
-        r"(?:linux-i686\.tar\.bz2|linux-x86_64\.tar\.bz2|linux-aarch64\.tar\.bz2|mac\.dmg|win32\.installer\.exe|win64\.installer\.exe|win64-aarch64\.installer\.exe))$"
+        r"^(/firefox/nightly/latest-(mozilla-central|try)-l10n/"
+        r"firefox-\d+\.0a1\.:lang\."
+        r"(?:"
+        r"linux-(i686|x86_64|aarch64)\.tar\.(bz2|xz)|"
+        r"mac\.dmg|"
+        r"win(32|64|64-aarch64)\.installer\.exe"
+        r"))$"
     ),
 }
 
 _BOUNCER_PATH_REGEXES_PER_ALTERNATIVE_PACKAGE_FORMAT = {
     **_BOUNCER_PATH_REGEXES_PER_PRODUCT_DEFAULT,
     "firefox-nightly-msi-latest-ssl": (
-        r"^(/firefox/nightly/latest-(mozilla-central|try)/firefox-\d+\.0a1\.en-US\." r"(?:win32\.installer\.msi|win64(?:|-aarch64)\.installer\.msi))$"
```

Comment:
Please move the changes like this (and many others further down) that are unrelated to xz support to their own PR. This helps reduces noise here, and ensures that a backout of xz support doesn't back this out, and vice versa.